### ISSUE TITLE
Make WASI preopened_dir more configurable

### ISSUE THREAD

**Issue by Mossaka** [containerd/runwasi#413]
I believe this is the way it's done today but in the case of the OCI layers I don't think we need these permissions.  Any thoughts on how we can make this more configurable?  It feels like this is where a OCI artifact with configuration that had info like this in it would come into play

_Originally posted by @jsturtevant in https://github.com/containerd/runwasi/pull/401#discussion_r1396414921_

**Comment by lengrongfu** [comment]
@Mossaka Can you give me some suggestions? 

`preopened` should how to config, we should use env or flags or config file to config it? now look like need to change https://github.com/containerd/rust-extensions this project.

**Comment by Mossaka** [comment]
I think the discussion from that PR was that we may add a config file as a OCI layer, but we haven't figured out if that's the route we want to go with. ping @jsturtevant for more context here.

**Comment by lengrongfu** [comment]
cc @justincormack  can you provide more info?

**Comment by jsturtevant** [comment]
@lengrongfu I don't think we have a clear idea of how we plan to handle this at this point.  This is on the table for discussion at the OCI sub-group in the wg-wasm cncf group.  If you have some thoughts or would like to listen in we are meeting next week Tue Feb 27th 8:00am - 9:00am (PST) (search for wasm on https://www.cncf.io/calendar/)

**Comment by TheRayquaza** [comment]
any update on this ?

I think we can even extend that issue to configuring the permissions including `allow_tcp`, `allow_udp` and so on
